shared/modules/buffer-utils.js

```diff
@@ -0,0 +1,16 @@
+import { toBuffer as ethUtilToBuffer, isHexString } from 'ethereumjs-util';
+
+/**
+ * Returns a buffer from the provided input, via ethereumjs-util.toBuffer but
+ * additionally handling non hex strings. This is a failsafe as in most cases
+ * we should be primarily dealing with hex prefixed strings with this utility
+ * but we do not want to break the extension for users.
+ * @param {import('ethereumjs-util').ToBufferInputTypes | string} input
+ * @returns {Buffer}
+ */
+export function toBuffer(input) {
+  if (typeof input === 'string' && isHexString(input) === false) {
```

Comment:
What is the `typeof` check meant to catch?

Reply:
this method will work with any input type that `ethereumjs.util` supports including BN, buffer, arrays of numbers, etc. But it only works with 0x prefixed hex strings, not plain strings. So this check is checking that if this is a string and not a hexstring then we just user Buffer.from

Reply:
Ah I see, makes sense. I didn't realize so many other types were supported.